scripts/customEventHooks.lua

```diff
@@ -2,64 +2,194 @@ local customEventHooks = {}
 
 customEventHooks.validators = {}
 customEventHooks.handlers = {}
+customEventHooks.scriptID = {
+	handlers = {},
+	validators = {},
+	generatedScriptIDs = {}
+}
+
+function customEventHooks.generateScriptID(filePath)
+	if not string.match(filePath, "%S") then 
+		return nil
+	end
+
+	if customEventHooks.scriptID.generatedScriptIDs[filePath] then
+		return customEventHooks.scriptID.generatedScriptIDs[filePath]
+	end
+
+	local seed = 0
+	for i = 1, #filePath do
+	  local charCode = string.byte(filePath:normalizePath(), i)
+	  seed = seed + charCode
+	end
+	math.randomseed(seed)
+
+    local template = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
```

Comment:
does the template need to be a local variable to the function?

Reply:
we can probably move it to the start of the file, theirs no reason for this to be a local in the func